scripts/check_frontend_test_coverage.py

```diff
@@ -0,0 +1,213 @@
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import os
+import shutil
+import subprocess
+import sys
+
+import python_utils
+
+
+def change_git_branch(branch):
+    """Changes git branch.
+
+    Args:
+      branch: string. Name of the branch to be changed.
+    """
+    task = subprocess.Popen(['git', 'checkout', branch])
+
+    task.communicate()
+    task.wait()
+
+    if branch == 'develop':
+        # This line is executed because CircleCI keeps the commits from PR
+        # branch when changing branchs. Ref ->
+        # (https://discuss.circleci.com/t/changing-git-branches-causes-
+        # commits-from-one-branch-to-carry-over-to-the-next/13562)
+        reset = subprocess.Popen(['git', 'reset', '--hard', 'origin/develop'])
+
+        reset.communicate()
+        reset.wait()
+
+
+def run_frontend_tests_script():
+    """Run the frontend tests script using subprocess."""
+    task = subprocess.Popen(['python', '-m', 'scripts.run_frontend_tests'])
+
+    task.communicate()
+    task.wait()
+
+
+def create_tmp_folder():
+    """Creates a temporary folder."""
+    if not os.path.exists('./tmp'):
+        os.mkdir('./tmp')
+
+
+def delete_tmp_folder():
+    """Delete the temporary folder."""
+    if os.path.exists('./tmp'):
+        shutil.rmtree('./tmp')
+
+
+def filter_lines(line):
+    """Check if the line has file path or total lines or covered lines
+        of the test.
+
+    Args:
+        line: str. A line from lcov file.
+
+    Returns:
+        True or false.
+    """
+    return bool(line.find('SF') >= 0 or line.find('LH') >= 0 or
+                line.find('LF') >= 0)
+
+
+def get_test_file_name(test_path):
+    """Get the file name from the absolute path.
+
+    Args:
+        test_path: string. The file's absolute path.
+
+    Returns:
+        file_name(str). The file name.
+    """
+    return test_path[test_path.rindex('/') + 1:]
+
+
+def get_lcov_file_tests(file_path):
+    """Get all tests from a lcov file, and filters it to return only:
+    - File path
+    - File total lines
+    - File covered lines
+
+    Args:
+      file_path: string. The path of lcov file.
+
+    Returns:
+      An array with all tests filtered, including ony important data for the
+      diff.
+    """
+    with python_utils.open_file(file_path, 'r') as f:
+        tests_array = f.read().split('end_of_record')
+        tests_array_filtered = []
+
+        for test in tests_array:
+            lines = [line for line in test.splitlines() if filter_lines(line)]
+            if len(lines) > 0:
+                tests_array_filtered.append(lines)
+
+        return tests_array_filtered
+
+
+def build_tests_fully_coverage_dict():
+    """Build a dict with only fully covered files from develop branch.
+
+    Returns:
+        A dict containing file path, total lines and covered lines of each
+        tested file.
+    """
+    coverage_dict = {}
+    file_path = './tmp/dev-lcov.info'
```

Comment:
This should be made into a global variable.

Reply:
Done